rpcs3/Emu/System.cpp

```diff
@@ -3318,4 +3324,43 @@ bool Emulator::IsVsh()
 	return g_ps3_process_info.get_cellos_appname() == "vsh.self"sv;
 }
 
+void Emulator::SaveSettings(const std::string& settings, const std::string& title_id)
+{
+	std::string config_name;
+
+	if (title_id.empty())
+	{
+		config_name = fs::get_config_dir() + "/config.yml";
+	}
+	else
+	{
+		config_name = rpcs3::utils::get_custom_config_path(title_id);
+	}
+
+	// Save config atomically
+	fs::pending_file temp(config_name);
+	temp.file.write(settings.c_str(), settings.size());
+	if (!temp.commit())
+	{
+		sys_log.error("Could not save config to %s (error=%s)", config_name, fs::g_tls_error);
```

Comment:
Add a system notification message as well. If the user has fullscreen, they won't know that settings didn't apply.

Reply:
Idk. why would it fail to save in the first place?
Seems like an arbitrary point to add a localized notification...

Reply:
Well, if we notify users when the setting saves then they will be more inclined to realize that settings have not saved if they close the emu while something is runing or they exit out of the guide menu without saving. Otherwise i agree

Reply:
> Idk. why would it fail to save in the first place? Seems like an arbitrary point to add a localized notification...

You're the one that added the check. If its unexpected behavior, throw.

Reply:
Btw, on the localized notification thing - does it need to be localized? I mean, you can just pass a string to the notifications, right?

Reply:
I didn't add anything. I just moved the code.
Also, what would a user faced message achieve if I'm not an english speaker and some random text appears for 3 seconds?